code/modules/antagonists/wizard/equipment/soulstone.dm

```diff
@@ -318,7 +322,18 @@
 		return TRUE
 
 	to_chat(user, "[span_userdanger("Capture failed!")]: The soul has already fled its mortal frame. You attempt to bring it back...")
-	INVOKE_ASYNC(src, PROC_REF(get_ghost_to_replace_shade), victim, user)
+
+	var/datum/component/ghost_poll/poll = AddComponent(/datum/component/ghost_poll, \
+		ignore_key = POLL_IGNORE_SHADE, \
+		job_bans = ROLE_CULTIST, \
+		title = "A shade" \
+	)
+
+	RegisterSignal(poll, COMSIG_GHOSTPOLL_CONCLUDED, PROC_REF(on_poll_concluded))
+
+	initiate_ref = WEAKREF(user)
+	victim_ref = WEAKREF(victim)
```

Comment:
...Because if you make it a callback you don't have to worry about tracking these weakrefs, you can just pass it with the callback

```dm
var/datum/callback/on_poll_success = CALLBACK(src, PROC_REF(on_poll_concluded), user, victim)
```

Reply:
How would I get the candidate?

Reply:
When you invoke it you can just pass the candidate in its arguments and I believe it'll concatenate it with the arguments you supply when making the cb. 

```dm
on_poll_success.Invoke(candidate)
```
...Will (should?) result in the proc being called with the arguments `(mob/user, mob/victim, mob/candiate)`.